core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -396,38 +395,64 @@ class CassandraJournal(cfg: Config) extends AsyncWriteJournal
     def hasTagsColumn(row: Row): Boolean =
       hasColumn("tags", row, _hasTagsColumn, updateTagsColumnCache)
 
-    def deserializeEvent(row: Row): Any = {
-      val event = serialization.deserialize(
-        Bytes.getArray(row.getBytes("event")),
-        row.getInt("ser_id"),
-        row.getString("ser_manifest")
-      ).get
-
-      if (hasMetaColumns(row)) {
-        row.getBytes("meta") match {
-          case null =>
-            event // no meta data
-          case metaBytes =>
-            // has meta data, wrap in EventWithMetaData
-            val metaSerId = row.getInt("meta_ser_id")
-            val metaSerManifest = row.getString("meta_ser_manifest")
-            val meta = serialization.deserialize(
-              Bytes.getArray(metaBytes),
-              metaSerId,
-              metaSerManifest
-            ) match {
-                case Success(m) => m
-                case Failure(_) =>
-                  // don't fail replay/query because of deserialization problem with meta data
-                  // see motivation in UnknownMetaData
-                  UnknownMetaData(metaSerId, metaSerManifest)
-              }
-            EventWithMetaData(event, meta)
+    def deserializeEvent(row: Row)(implicit ec: ExecutionContext): Future[Any] = try {
+
+      def meta: OptionVal[AnyRef] = {
```

Comment:
could be a val?

Reply:
`def` because called from inside the future map